Tools/CMake/AMReXParallelBackends.cmake

```
@@ -208,8 +208,9 @@ if (AMReX_HIP)
 
    # Link to hiprand -- must include rocrand too
    find_package(rocrand REQUIRED CONFIG)
+   find_package(rocprim REQUIRED CONFIG)
    find_package(hiprand REQUIRED CONFIG)
-   target_link_libraries(amrex PUBLIC hip::hiprand roc::rocrand)
+   target_link_libraries(amrex PUBLIC hip::hiprand roc::rocrand roc::rocprim ${HIP_LIBRARIES})
```

Let's try w/o the `HIP_LIBRARIES` for now, CI seems to have trouble with those.

---

Or we try adding those flags only to C++:
```suggestion
   target_link_libraries(amrex PUBLIC hip::hiprand roc::rocrand roc::rocprim $<$<COMPILE_LANGUAGE:CXX>:${HIP_LIBRARIES}>)
```